### PR TITLE
Add support for Renesas RCAR H3

### PR DESCRIPTION
**Pull request by lorc**
I have added new platform - plat-rcar.

There are only minimal peripherals support at this time. Only debug UART is working. Anyways, it was tested on renesas salvator-x board. All tests from xtest suite have passed successfully.

### CODE REVIEW

File: core/drivers/scif.c [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+#include <io.h>
+#include <util.h>
+#include <drivers/scif.h>
+
+#define SCIF_SCFSR	(0x10)
+#define SCIF_SCFTDR	(0x0C)
+#define SCIF_SCFCR	(0x18)
+#define SCIF_SCFDR	(0x1C)
+
+#define SCFSR_TEND	(1 << 6)
```
Comment (jenswi-linaro) [inline core/drivers/scif.c]: Please use the `BIT` or `BIT32` macro from `<util.h>`

File: core/drivers/scif.c [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+void scif_uart_init(vaddr_t base)
+{
+	/* Bootloader should initialize device for us */
+	scif_uart_flush(base);
+}
+
+void scif_uart_putc(int ch, vaddr_t base)
+{
+	/* Wait until there is space in the FIFO */
+	while ((read16(base + SCIF_SCFDR) >> 8) >= 15)
```
Comment (jenswi-linaro) [inline core/drivers/scif.c]: `8` and `15` looks like magic number, please define them.

File: core/arch/arm/plat-rcar/platform_config.h [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+ */
+#define CFG_TEE_RAM_PH_SIZE	CFG_TEE_RAM_VA_SIZE
+#define CFG_TEE_RAM_START	(TZDRAM_BASE + 0x00100000)
+#define CFG_TA_RAM_START	ROUNDUP((CFG_TEE_RAM_START + \
+					CFG_TEE_RAM_VA_SIZE), \
+					CORE_MMU_DEVICE_SIZE)
+#define CFG_TA_RAM_SIZE		ROUNDDOWN((TZDRAM_SIZE - CFG_TEE_RAM_VA_SIZE), \
+					  CORE_MMU_DEVICE_SIZE)
+#ifdef CFG_WITH_LPAE
+#define MAX_XLAT_TABLES		8
```
Comment (jenswi-linaro) [inline core/arch/arm/plat-rcar/platform_config.h]: Is this needed?

File: core/arch/arm/plat-rcar/conf.mk [hunk 4]
```diff
@@ -0,0 +1,28 @@
+PLATFORM_FLAVOR ?= h3
+PLATFORM_FLAVOR_$(PLATFORM_FLAVOR) := y
```
Comment (jforissier) [inline core/arch/arm/plat-rcar/conf.mk]: Remove this line (line 2). The variable is now set automatically in the main Makefile (I noticed some platforms still have such a line, I will create a PR to clean them up).

**Comment by jforissier** [comment]
A few more comments:
- Could you add a commit to update .travis.yml to build for the new platform? Similar to this one: https://github.com/OP-TEE/optee_os/commit/66d2f371980c2c3d057664b8b5439f15ee76709b.
- Could you also update `README.md` section "Plaftorms supported"?
- There is also `MAINTAINERS.md` that you may update if you're OK with being pinged from time to time, such as when we make a new release of OP-TEE and we'd like to have Tested-by's for as many platforms as possible. If you don't want to appear as a maintainer it's OK too.

File: core/arch/arm/plat-rcar/platform_config.h (hunk 3 above)
Comment (lorc) [inline core/arch/arm/plat-rcar/platform_config.h]: You are talking about CFG_TEE_RAM_START definition?
I'm not sure. Secure RAM begins at TZDRAM_BASE, but first 0x100000 bytes are used by ARM Trusted FW.  Probably we can move TZDRAM_BASE by this 0x100000 bytes and simplify calculations at line 77. But I don know if it is right thing. What do you think?

Comment (jenswi-linaro) [inline core/arch/arm/plat-rcar/platform_config.h]: No, I'm talking about `MAX_XLAT_TABLES`. Do you need to use something different from default?

Comment (lorc) [inline core/arch/arm/plat-rcar/platform_config.h]: Yep, you are right. It works with default value.

**Comment by jenswi-linaro** [comment]
`Reviewed-by: Jens Wiklander <jens.wiklander@linaro.org>`

**Comment by lorc** [comment]
Should I rebase my patches on current master?

**Comment by jenswi-linaro** [comment]
When you do rebase please add my RB tag.
`Reviewed-by: Jens Wiklander <jens.wiklander@linaro.org>`

Since Jerome has had some comments you could wait until he's acking too.

**Comment by jforissier** [comment]
@lorc yes, please rebase and add Jens' Reviewed-by: to each one of the three commits, and you may also add mine:
`Reviewed-by: Jerome Forissier <jerome.forissier@linaro.org>`

**Comment by jforissier** [comment]
@lorc thanks for your contribution!

**Comment by jbech-linaro** [comment]
Cool, indeed @lorc, thanks for taking the time to contribute.

**Comment by lorc** [comment]
@jforissier , @jbech-linaro , it's nothing. If things will go fine, there will be much more from my side.

**Pull request merged**